src/snowflake/cli/plugins/nativeapp/version/version_processor.py

```diff
@@ -46,18 +43,18 @@ def check_index_changes_in_git_repo(
 
         # Check if the repo has any changes, including untracked files
         if repo.is_dirty(untracked_files=True):
-            print("Changes detected in your git repository!")
+            cc.warning("Changes detected in your git repository!")
```

Comment:
nit: maybe something like "Changes detected in the git repository (rerun with --skip-git-check to ignore)"? Would be nice to tell the user how to get around this check if they don't expect this. I do like how the git CLI essentially teaches you how to use it as you interact with the output.